### DISCUSSION TITLE
Find the matrix of bond length both geometric and length of bond

### DISCUSSION THREAD

**Discussion by vs140580** [rdkit/rdkit#5901] [Development]
Kind help someone with a code to find the
1) Matrix of Bond length actual between two atoms in the molecule after explicit hydrogen given smiles file
2) Matrix of  geometric distance  between two atoms in the molecule after explicit hydrogen given smiles file

Kind help

Not the topological bond length

If possible kind help to the correct code page if already exists Please kind help
Using RdKit

**Comment by greglandrum** [comment]
Hi @vs140580
If you construct a molecule from SMILES then you normally don't have coordinates available. This means that you can't calculate geometric distances.

Can you provide a bit more information about what you are trying to do?

**Reply by vs140580** [reply]
Respected sir,
I want to input a molecule to python in some format and I want the code to return the matrix with bond lengths between atoms and another geometric length between atoms. In the hydrogen depleted molecule of inputted molecule. Can I get how we can do this 
Kind help with a code
How to get the geometric distances matrix of a molecule from a python code for the hydrogen depleted form of the molecule

**Comment by vs140580** [comment]
Respected sir,
I want to input a molecule to python in some format and I want the code to
return the matrix with bond lengths between atoms and another geometric
length between atoms. In the hydrogen depleted molecule.

**Reply by greglandrum** [reply]
If you have read in a molecule which has 3D coordinates then you can use `Chem.Get3DDistanceMatrix()` to calculate the distances between the atoms.

File formats that the RDKit supports which have 3D coordinates include mol and sdf. The RDKit Getting Started in Python guide explains how to read from these formats.
